lib/bolt/node_uri.rb

```diff
@@ -7,15 +9,13 @@ def initialize(string)
     def parse(string)
       case string
       when %r{^(ssh|winrm|pcp)://.*:\d+$}
-        URI(string)
-      when %r{^pcp://}
-        URI(string)
-      when %r{^(ssh|winrm)://}
-        uri = URI(string)
+        Addressable::URI.parse(string)
+      when %r{^(ssh|winrm|pcp)://}
+        uri = Addressable::URI.parse(string)
```

Comment:
Looks good, but investigating if this can be further collapsed / simplified.

Reply:
I have a different PR in progress to further simplify. The port vs non-port can be collapsed, but we still have to differentiate between when a scheme is specified vs not.

Reply:
This works fine:

```ruby
    def parse(string)
      string = "ssh://#{string}" if !string.match?(%r{^(ssh|winrm|pcp)://.*$})
      uri = Addressable::URI.parse(string)
      uri.port = 5985 if (uri.scheme == 'winrm' && uri.port.nil?)

      uri
    end
```